sdk/core/core/inc/az_span.h

```diff
@@ -54,10 +54,39 @@ AZ_NODISCARD AZ_INLINE int32_t az_span_size(az_span span) { return span._interna
 
 /********************************  CONSTRUCTORS */
 
+/**
+ * @brief Returns an #az_span over a byte buffer.
+ *
+ * @param[in] ptr The memory address of the 1st byte in the byte buffer.
+ * @param[in] size The number of total bytes in the byte buffer.
+ * @return The "view" over the byte buffer.
+ */
+// Note: If you are modifying this method, make sure to modify the non-inline version in the
+// az_span.c file as well, and the _az_ version right below.
+#ifdef AZ_NO_PRECONDITION_CHECKING
+AZ_NODISCARD AZ_INLINE az_span az_span_init(uint8_t* ptr, int32_t size)
+{
+  return (az_span){ ._internal = { .ptr = ptr, .size = size } };
+}
+#else
+AZ_NODISCARD az_span az_span_init(uint8_t* ptr, int32_t size);
+#endif // AZ_NO_PRECONDITION_CHECKING
+
+// Same function as above, only for use in AZ_SPAN_FROM_BUFFER macro.
+// So that the users don't get warnings in their code when they use the macro:
+// C4221: nonstandard extension: 'ptr': cannot be initialized using address of automatic variable
+// C4204: nonstandard extension: non-constant aggregate initializer
+AZ_NODISCARD AZ_INLINE az_span _az_span_init(uint8_t* ptr, int32_t size)
+{
+  return (az_span){ ._internal = { .ptr = ptr, .size = size } };
+}
+
```

Comment:
I wish we didn't have to do this. Can we workaround the warning in a different way that doesn't require code duplication? Who is seeing these warnings because when I build locally, I didn't. Is  this a side effect of removing `<_az_cfg.h>`?

In the future, please split changes like these out into separate PRs.

Reply:
This is absolutely a side effect of removing az_cfg.h, where we disable a bunch of nonsensical warnings, but we can't require our customers to include az_cfg, or to disable them.
The warnings that are explicitly disabled in samples, do not come from the SDK code, but they are from the sample code.

Reply:
Regarding code duplication, I talked with Jeff, I will change the macro to simply use az_span_init(), which will check preconditions when built with precondition support, but he is fine with it, and thinks it even might help to spot some bug.